src/scenario/data.h

```diff
@@ -128,19 +153,7 @@ typedef struct {
 #define DEMAND_CHANGE_LEGACY_IS_FALL -9999
 
 typedef struct {
-    struct win_criteria_t population;
-    struct win_criteria_t culture;
-    struct win_criteria_t prosperity;
-    struct win_criteria_t peace;
-    struct win_criteria_t favor;
-    struct {
-        int enabled;
-        int years;
-    } time_limit;
-    struct {
-        int enabled;
-        int years;
-    } survival_time;
+    win_criteria_t goals[MAX_WIN_CRITERIA];
     int milestone25_year;
     int milestone50_year;
     int milestone75_year;
```

Comment:
This might be hard to calculate for non-standard criteria.

Might be best to leave this sort of favor adjustment (For non-standard goals at least) to be handled by map makers and scenario events. (Once that is added.)